tests/test_herbstclient.py

```diff
@@ -0,0 +1,43 @@
+import subprocess
+import os
+import re
+import pytest
+
+HC_PATH = os.path.join(os.path.abspath(os.environ['PWD']), 'herbstclient')
+
+
+@pytest.mark.parametrize('hlwm_mode', ['never started', 'sigterm', 'sigkill'])
+@pytest.mark.parametrize('hc_parameter', ['true', '--wait'])
+def test_herbstclient_recognizes_hlwm_not_running(hlwm_spawner, x11, hlwm_mode, hc_parameter):
+    if hlwm_mode == 'never started':
+        pass
+    else:
+        # start hlwm and kill it in order to leave X in some inconsistent state
+        hlwm_proc = hlwm_spawner()
+        # ping that herbstclient still works here
+        hc = subprocess.run([HC_PATH, 'echo', 'ping'],
+                            stdout=subprocess.PIPE,
+                            text=True,
```

Comment:
Given my previous stance on it, I'm not really in a position to argue for supporting older Python versions, but note that using `text=True` here is the only part the would make us incompatible with Python 3.6 (i.e., require 3.7). `text=True` is a convenience alias for `universal_newlines=True`, so we *could* use that to stay compatible with 3.6 for a while longer.

Reply:
I didn't know this and thus changed it to `universal_newlines=True`.